### PR TITLE
Implementation of encryption keys features, Issue #10

### PR DESCRIPTION
**Pull request by ylimiky**
#### What does this PR do?
This PR implements feature #10 by passing key parameter from client to the methods which need keys. Keys are asked from the user in the client and passed to the methods that need them. If method requires keys and user doesn't give them, error will be thrown.

#### Where should the reviewer start?
Looking at the changes to the client and to the API.

#### How should this be manually tested?
Using the client and not giving a key to a method which requires key will result in error message. Leaving the key empty for a method which doesn't require key causes no error.

#### What are the relevant tickets?
Closes issue #10

### CODE REVIEW

File: JEasyCrypto/src/easycrypto/CyrMethod.java [hunk 1]
```diff
@@ -46,6 +51,18 @@ public Result decrypt(String toDecrypt) {
         }
 		return new Result(ResultCode.ESuccess, result);
 	}
+	
+	//empty methods for interface
+	
+		@Override
```
Comment (a-rmz) [inline JEasyCrypto/src/easycrypto/CyrMethod.java]: There is a weird indentation going on here... one tab extra.

File: JEasyCrypto/src/easycrypto/ReverseMethod.java [hunk 2]
```diff
@@ -22,4 +27,16 @@ public String method() {
 		return "reverse";
 	}
 
+	//empty methods for interface
+	
+		@Override
```
Comment (a-rmz) [inline JEasyCrypto/src/easycrypto/ReverseMethod.java]: Same indentation thing here.

**Comment by a-rmz** [comment]
Good! 👍 Now just there are some merge conflicts to fix before merging.

**Comment by ylimiky** [comment]
Hey, can you merge this? :)

**Pull request merged**